### PR TITLE
Linalg refactor

### PR DESCRIPTION
**Pull request by rocky**
These were the remaining ListExpression items I didn't have the energy to go over in the first round. 
Here I also sort the builtin class names and move the Numeric Distance functions into Distance and Similarity measures as it is described and organized in https://reference.wolfram.com/language/guide/DistanceAndSimilarityMeasures.html

This change alone is big enough for its own PR.

In general a PR is no bigger than what I can tolerate in a single session that I have time for. I am also hoping that it also limits the amount of effort to double check this as well. 

To reduce confusion it may be useful to look at the 3 commits individually, especially commit 2 and then commit 3.

<a href="https://gitpod.io/#https://github.com/Mathics3/mathics-core/pull/339"></a>

### CODE REVIEW

File: mathics/builtin/distance/numeric.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    """
+
+    summary_text = "Canberra distance"
+
+    def apply(self, u, v, evaluation):
+        "CanberraDistance[u_, v_]"
+        t = _norm_calc("Subtract", u, v, evaluation)
+        if t is not None:
+            return Expression(
+                "Total",
```
Comment (TiagoCavalcante) [inline mathics/builtin/distance/numeric.py:89]: SymbolTotal

File: mathics/builtin/distance/numeric.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+    summary_text = "cosine distance"
+
+    def apply(self, u, v, evaluation):
+        "CosineDistance[u_, v_]"
+        dot = _norm_calc("Dot", u, v, evaluation)
+        if dot is not None:
+            return Expression(
+                SymbolSubtract,
+                1,
```
Comment (TiagoCavalcante) [inline mathics/builtin/distance/numeric.py:148]: Integer1

File: mathics/builtin/numbers/linalg.py [hunk 3]
```diff
@@ -178,202 +109,303 @@ def apply(self, a, b, evaluation):
...
+        eigenvalues = [(from_sympy(v), c) for (v, c) in eigenvalues]
 
+        # Sort the eigenvalues by their sort key
+        eigenvalues.sort(key=lambda v: v[0].get_sort_key())
 
-class PseudoInverse(Builtin):
-    """
+        eigenvalues = [v for (v, c) in eigenvalues for _ in range(c)]
+
+        return Expression("List", *eigenvalues)
```
Comment (TiagoCavalcante) [inline mathics/builtin/numbers/linalg.py:254]: ListExpression

**Review by TiagoCavalcante — APPROVED** [review]
LGTM

File: mathics/builtin/distance/numeric.py (hunk 1 above)
Comment (rocky) [reply mathics/builtin/distance/numeric.py]: And other places in this fine. Done now.

**Pull request merged**